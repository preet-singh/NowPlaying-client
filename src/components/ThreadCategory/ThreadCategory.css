@media screen and (min-width: 320px){

    .thread-category > h3 {
        font-size: 32px;
    }
    
}