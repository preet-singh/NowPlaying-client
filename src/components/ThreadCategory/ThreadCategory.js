import React from 'react';
import {Link, withRouter} from 'react-router-dom';
import '../ThreadsList/ThreadsList.css';
import ThreadsList from '../ThreadsList/ThreadsList';
import UserContext from '../../utils/context';
import ThreadItem from '../ThreadItem/ThreadItem';



class ThreadCategory extends React.Component {

  static contextType = UserContext;
  constructor(props) {
    super(props);
    this.state = {
    }
  }

  mostPopular = () => {
    if(!this.context.categoryItems) {
      return 'No threads yet!'
    }
    let movies = this.context.categoryItems;
    movies.sort((a,b) => (a.imdb_rating > b.imdb_rating));
    let amount = 3;
    if (movies.length < 3) { amount = movies.length }
    let results = [];
    for (let i=0;i < amount; i++) {
      results.push(<ThreadItem details={movies[i]} key={movies[i].id} />);
    }
    return results;
  }
  
  checkIfHome = () => {
    if (this.props.location.pathname === '/') {
      return <Link to={`category/${this.context.categoryID}`}>see more</Link>
    }
  }


  render(){
    return (
      <div className='threads-list'>
        <h3>POPULAR MOVIES</h3>
        <h4><Link to="/category/1">(see all)</Link></h4>
          <div className="threads-list-container">
            {this.mostPopular()}
          </div>
      </div>
    )
  }
}

export default withRouter(ThreadCategory);