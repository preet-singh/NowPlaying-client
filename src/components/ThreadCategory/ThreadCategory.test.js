import React from 'react';
import ThreadCategory from './ThreadCategory.js';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe(`ThreadCategory component`, () => {

  it('renders the ThreadCategory', () => {
    const wrapper = shallow(<ThreadCategory />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

})