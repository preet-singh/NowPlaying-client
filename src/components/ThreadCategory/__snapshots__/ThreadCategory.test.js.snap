// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThreadCategory component renders the ThreadCategory 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
